no-std support?

Hey! 

Thanks for an awesome repo. I have a Pinocchio-based Solana contract where I want to use this crate. 
Problem is that my whole program is no-std, while brine-fp requires std feature to be enabled. 

Without `[std]` build fails with an error:


```
    Blocking waiting for file lock on package cache
    Blocking waiting for file lock on package cache
   Compiling crunchy v0.2.4
   Compiling hex v0.4.3
   Compiling byteorder v1.5.0
   Compiling uint v0.9.5
   Compiling brine-fp v0.2.1
error: cannot find macro `format` in this scope
   --> src/unsigned.rs:287:9
    |
287 |         format!("{}.{:0>width$}", whole, decimals, width = 18)
    |         ^^^^^^

error: cannot find macro `format` in this scope
   --> src/signed.rs:121:9
    |
121 |         format!("{}{}", sign, self.value.to_string())
    |         ^^^^^^

error[E0412]: cannot find type `String` in this scope
   --> src/unsigned.rs:278:32
    |
278 |     pub fn to_string(&self) -> String {
    |                                ^^^^^^ not found in this scope

error[E0412]: cannot find type `String` in this scope
   --> src/signed.rs:119:32
    |
119 |     pub fn to_string(&self) -> String {
    |                                ^^^^^^ not found in this scope

For more information about this error, try `rustc --explain E0412`.
error: could not compile `brine-fp` (lib) due to 4 previous errors

```

Issue is trivial to fix, I can open PR to do that

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -6,6 +6,9 @@
 #![allow(clippy::manual_range_contains)]
 #![allow(clippy::reversed_empty_ranges)]
 
+#[cfg(not(feature = "std"))]
+extern crate alloc;
+
 use uint::construct_uint;
 
 construct_uint! {

--- a/src/signed.rs
+++ b/src/signed.rs
@@ -1,6 +1,9 @@
 use super::unsigned::UnsignedNumeric;
 use core::ops::{Add, Sub, Mul, Div};
 
+#[cfg(not(feature = "std"))]
+use alloc::{format, string::String};
+
 // Based on the following implementations:
 // https://github.com/solana-labs/solana-program-library/blob/v2.0/libraries/math/src/precise_number.rs
 // https://github.com/StrataFoundation/strata/blob/master/programs/spl-token-bonding/src/signed_precise_number.rs

--- a/src/unsigned.rs
+++ b/src/unsigned.rs
@@ -4,6 +4,9 @@ use super::InnerUint;
 use core::convert::*;
 use core::ops::{Add, Sub, Mul, Div};
 
+#[cfg(not(feature = "std"))]
+use alloc::{format, string::String};
+
 // Based on the following implementations:
 // https://github.com/solana-labs/solana-program-library/blob/v2.0/libraries/math/src/precise_number.rs
 // https://github.com/StrataFoundation/strata/blob/master/programs/spl-token-bonding/src/precise_number.rs

